src/recon_buildblock/KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData.cxx

```diff
@@ -0,0 +1,1919 @@
+                                             exp(-square(distance[dz][dy][dx]/grid_spacing.x ())/(2*square(get_sigma_dp())));
+                                        }
+
+                            }
+                            else{
+                                kPET=1;
+
+                            }  // the following "pnkernel" is the normalisation of the kernel
+                                    kAnatomical1=exp(-square(((*anatomical1_sptr)[z][y][x]-(*anatomical1_sptr)[z+dz][y+dy][x+dx])/kSD*sigma_m)/2)*
+                                                 exp(-square(distance[dz][dy][dx]/grid_spacing.x ())/(2*square(sigma_dm)));
+
+                                    pnkernel+=kPET*kAnatomical1;
+
+                                    kImage[z][y][x] += kAnatomical1*kPET*Image[z+dz][y+dy][x+dx];//
+
+                                   }
+                                    if(current_estimate[z][y][x]==0){
+                                         continue;}
+                                     kImage[z][y][x]= kImage[z][y][x]/pnkernel;
+                                     pnkernel=0;
+                       }
+                    }
+              }
+
+}
+
+
+template<typename TargetT>
+void
+KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData<TargetT>::
+compute_sub_gradient_without_penalty_plus_sensitivity(TargetT& gradient,
+                                                      const TargetT &current_estimate,
+                                                      const int subset_num)
+{
+
+
+    set_subiter_num (this->subiter_num+=1);
+
+    TargetT& kImage= *current_estimate.get_empty_copy();
+
+    if(this->num_non_zero_feat==1){
+        fast_compute_kernelised_image (kImage, *current_estimate.clone(), current_estimate);
+                                    }
+    else{
+      compute_kernelised_image(kImage, *current_estimate.clone(),
+                               current_estimate);
+    }
+if((get_subiter_num ()-1)%this->get_num_subsets()==0){
+//        this->current_kimage_filename =
+//          boost::str(boost::format(this->kernelised_output_filename_prefix) %get_subiter_num ());
+
+    char itC[10];
+    sprintf (itC, "%d", get_subiter_num ()-1);
+    std::string it=itC;
+    std::string us="_";
+    std::string k="_k.hv";
+    this->current_kimage_filename =this->kernelised_output_filename_prefix+us+it+k;
+std::cout<<this->current_kimage_filename<<std::endl;
+    write_to_file(this->current_kimage_filename,kImage);
+
+
+//    this->current_kimage_filename =  ;
+}
+
+
+    const std::string current_sensitivity_filename =
+      boost::str(boost::format(this->get_subsensitivity_filenames ()) % subset_num);
+
+    //if(get_subiter_num ()<=this->get_num_subsets()){
+         shared_ptr<TargetT> sens_sptr(read_from_file<TargetT>(current_sensitivity_filename));
```

Comment:
Yes, the sensitivity is calculated before reconstruction so both if you read or recalculate the sensitivity this works. However, I seem to remember that I did this because I was having problem to access the sensitivity from the code.
I am open to suggestion though...